common/src/node_id.rs

```diff
@@ -109,8 +109,8 @@ impl From<&NodeID> for ResponderId {
 }
 
 // This is needed for SCPNetworkState's NetworkState implementation.
-impl AsRef<ResponderId> for NodeID {
-    fn as_ref(&self) -> &ResponderId {
-        &self.responder_id
+impl AsRef<NodeID> for NodeID {
```

Comment:
Nit: This is unnecessary, and might clash with a default/blanket impl. Let's remove it